sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -90,18 +94,24 @@ public static void main(String[] args) throws Exception {
   public static void main(PipelineOptions options,
       Endpoints.ApiServiceDescriptor loggingApiServiceDescriptor,
       Endpoints.ApiServiceDescriptor controlApiServiceDescriptor) throws Exception {
-    ManagedChannelFactory channelFactory = ManagedChannelFactory.from(options);
+    ManagedChannelFactory channelFactory;
+    List<String> experiments = options.as(DataflowPipelineDebugOptions.class).getExperiments();
```

Comment:
<!--new_thread; commit:ae51d75eee0f8932a660a25ad45491f2e8e2762a; resolved:0-->
This will obviously have to go, breaking the dep

Reply:
I would be for elevating experiments to be a pipeline option within SDK core instead of breaking this chain.